test/e2e/network/dual_stack.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2edeploy "k8s.io/kubernetes/test/e2e/framework/deployment"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+// Tests for ipv6 dual stack feature
+var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("dualstack")
+
+	var cs clientset.Interface
+	var podClient *framework.PodClient
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+		podClient = f.PodClient()
+	})
+
+	ginkgo.It("should have ipv4 and ipv6 internal node ip", func() {
+		// TODO (aramase) can switch to new function to get all nodes
+		nodeList := framework.GetReadySchedulableNodesOrDie(cs)
+
+		for _, node := range nodeList.Items {
+			// get all internal ips for node
+			internalIPs := e2enode.GetAddresses(&node, v1.NodeInternalIP)
+
+			framework.ExpectEqual(len(internalIPs), 2)
+			// assert 2 ips belong to different families
+			framework.ExpectEqual(isIPv4(internalIPs[0]) != isIPv4(internalIPs[1]), true)
```

Comment:
After merging https://github.com/kubernetes/kubernetes/pull/79862 we will be able to replace here with `framework.ExpectNotEqual(isIPv4(internalIPs[0]), isIPv4(internalIPs[1]))`.
But it is fine at this time.